optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -185,29 +185,33 @@ def _calculate_categorical_distributions(
         search_space: CategoricalDistribution,
         parameters: _ParzenEstimatorParameters,
     ) -> _BatchedDistributions:
-        consider_prior = parameters.consider_prior or len(observations) == 0
+        choices = search_space.choices
+        n_choices = len(choices)
+        n_kernels = len(observations) + (parameters.consider_prior or len(observations) == 0)
 
         assert parameters.prior_weight is not None
         weights = np.full(
-            shape=(len(observations) + consider_prior, len(search_space.choices)),
-            fill_value=parameters.prior_weight / (len(observations) + consider_prior),
+            shape=(n_kernels, n_choices),
+            fill_value=parameters.prior_weight / n_kernels,
         )
 
-        if param_name in parameters.categorical_distance_func:
+        observed_indices = observations.astype(int)
+        if len(observed_indices) == 0:
+            # Do nothing.
+            pass
```

Comment:
Is there any reason that you added this logic?

Reply:
This was originally for avoiding an unnecessary nesting, but I replaced it with early-return.

Reply:
I wondered why this logic was additionally needed. I investigated and I realized it is for ensuring this vector to be 2-dimensional: `dists = np.array([[dist_func(choices[i], c) for c in choices] for i in used_indices])`. Thanks for explaining.